packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,516 @@
+                                  @"LANDSCAPE_LEFT" : @(EXOrientationLandscapeLeftLock),
+                                  @"LANDSCAPE_RIGHT" : @(EXOrientationLandscapeRightLock),
+                                  @"OTHER" : @(EXOrientationOtherLock),
+                                  @"ALL_BUT_UPSIDE_DOWN": @(EXOrientationAllButUpsideDownLock)
+                                  };
+  });
+  return strToOrientationLockDict;
+}
+
++ (NSDictionary *)getOrientationLockJSToStringDict
+{
+  static NSMutableDictionary*orientationLockToStrDict = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    orientationLockToStrDict = [[NSMutableDictionary alloc] init];
+    NSDictionary *strToOrientationLock = [EXScreenOrientationModule getStringToOrientationLockJSDict];
+    for(NSString *str in strToOrientationLock) {
+      NSNumber *wrappedOrientationLock = [strToOrientationLock objectForKey:str];
+      orientationLockToStrDict[wrappedOrientationLock] = str;
+    }
+  });
+  return orientationLockToStrDict;
+}
+
+- (UIInterfaceOrientationMask)orientationMask
+{
+  return [[EXScreenOrientationModule sharedRegistry] orientationMaskForAppId:defaultAppId];
+}
+
+- (void)setOrientationMask:(UIInterfaceOrientationMask)mask
+{
+  return [[EXScreenOrientationModule sharedRegistry] setOrientationMask:mask forAppId:defaultAppId];
+}
+
++ (EXScreenOrientationRegistry *)sharedRegistry
+{
+  return (EXScreenOrientationRegistry *)[UMModuleRegistryProvider getSingletonModuleForClass:[EXScreenOrientationRegistry class]];
+}
+
+
+
+// Will be called when this module's first listener is added.
+- (void)startObserving
+{
+  _hasListeners = YES;
+}
+
+// Will be called when this module's last listener is removed, or on dealloc.
+- (void)stopObserving
+{
+//  hasListeners = NO;
+//  if ([[UIDevice currentDevice] isGeneratingDeviceOrientationNotifications]){
+//    [[UIDevice currentDevice] endGeneratingDeviceOrientationNotifications];
+//  }
+}
+
+- (void)handleDeviceOrientationChange:(NSNotification *)notification
+{
+  UIDevice *device = notification.object;
+  UIInterfaceOrientation currentDeviceOrientation = [self UIDeviceOrientationToUIInterfaceOrientation:[device orientation]];
+  UIInterfaceOrientationMask orientationMask = [self orientationMask];
+  
+  // first check: phone only rotates if device orientation is in mask, second check: we should send event only if device didn't rotate to current screen orientation
+  if ([self doesOrientationMask:orientationMask containOrientation:currentDeviceOrientation] && _currentScreenOrientation != currentDeviceOrientation) {
+    _currentScreenOrientation = currentDeviceOrientation;
+    if (_hasListeners) {
+      [self handleScreenOrientationChange];
+    }
+  }
+}
+
+- (void)handleScreenOrientationChange
+{
+  EXOrientationLock orientationLock = [self orientationLockNativeToJS:[self orientationMask]];
+  [self->_eventEmitter sendEventWithName:@"expoDidUpdateDimensions" body:@{
```

Comment:
Why didn't `_eventEmitter` suffice?

Reply:
There was dispatch get main queue in this method earlier and I forgot to change it after removing.